script.js

```diff
@@ -1,12 +1,298 @@
+
+let bod = document.getElementsByTagName("BODY")[0];
+let main = document.createElement("div")
+main.setAttribute("class", "main")
+bod.appendChild(main)
+
+let pages = document.createElement("div")
+pages.setAttribute("class", "pages")
+bod.appendChild(pages)
+
+
+let showId;
+let episode;
+
+//Setup Function
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  searchBar()
+  searchFilter()
+  showsList()
+  showSelection()
+  getShows()
+  citation()
+}
+
+//Populates the page with the shows
+function getShows(){
+  const rootElem = document.getElementById("root");
+
+  let shows = getAllShows();
+  let allShows = shows.sort(function compare(a, b) {
+    if (a.name.toUpperCase() < b.name.toLocaleUpperCase()) { return -1 }
+    if (a.name.toUpperCase() > b.name.toUpperCase()) { return 1 }
+    return 0;
+  })
+
+  for(i=0; i<allShows.length; i++){
+    let mainDiv = document.createElement("div")
+    mainDiv.setAttribute("class", "show-card")
+
+    let title = `${allShows[i].name} `
+    let h2 = document.createElement("h2")
+    h2.setAttribute("id", allShows[i].id)
+    h2.addEventListener("click", presentEpisodes)
+    h2.innerHTML = title
+
+    let page = `${allShows[i].summary}`
+    let p = document.createElement("p")
+    p.setAttribute("id", "show-text")
+    p.innerHTML = page
+
+    let status = `Genre: ${allShows[i].genres} Status: ${allShows[i].status}`
+    let h4 = document.createElement("h4")
+    h4.setAttribute("class", "status")
+    h4.innerHTML = status
+
+    let headers = `Rating: ${allShows[i].rating.average} Runtime: ${allShows[i].runtime}`
+    let h5 = document.createElement("h5")
+    h5.setAttribute("class", "headers")
+    h5.innerHTML = headers
+
+    let image = document.createElement("img")
+    image.setAttribute("class", "show-images")
+    let img = allShows[i].image?.medium
+    image.src = img
+
+    rootElem.appendChild(mainDiv)
+    mainDiv.append(h2, image, p, h4, h5)
+  }
 }
 
-function makePageForEpisodes(episodeList) {
+//Fetch and Print all episodes to webpage (default parameter)
+function getEpisodes(showId){
+  const url = `https://api.tvmaze.com/shows/${showId}/episodes`;
+    fetch(url)
+    .then(function(response){
+        if(response.ok){
+            return response.json();
+        }
+        throw `${response.status} ${response.statusText}`;
+    })
+    .then(function(data) {
+    
+    //console.log(data)
+    const rootElem = document.getElementById("root");
+    rootElem.innerHTML = "";
+ 
+  for(i=0; i<data.length; i++){
+    let newDiv = document.createElement('div');
+    newDiv.setAttribute("class", "card");
+    rootElem.append(newDiv)
+
+    const {name, season, number, summary, image: {medium, original}} = data[i]
+    let season_Names = document.createElement("h3")
+    season_Names.setAttribute("class", "season-names")
+    let episodeCode = `S${season.toString().padStart(2, "0")}E${number.toString().padStart(2, "0")}`
```

Comment:
this line is a little long, maybe we could do a function here to handle the padding.. like

```
function formatEpisodeNumber(season, episode) {
   const padSeason = `S${season.toString().padStart(2, "0")}`
   const padEpisode = `E${episode().padStart(2, "0")}`
   return padSeason + padEpisode
}
```

? not necessary, just may be a lil cleaner